### PR TITLE
Add pyproject.toml for python packaging

### PR DESCRIPTION
**Pull request by JenySadadia**
Fixes https://github.com/kernelci/kernelci-core/issues/2283

With python 3.12 the support for setup.py is deprecated. Hence, update the packaging and distribution of 'kernelci' package by adding 'pyproject.toml' file.

### CODE REVIEW

**Comment by yurinnick** [comment]
nit: If you simply moving files into `scripts` directory, `git mv` will do it better, preserving a history and keeping diff clean.

**Comment by JenySadadia** [comment]
> nit: If you simply moving files into `scripts` directory, `git mv` will do it better, preserving a history and keeping diff clean.

Yes, that's true. But I didn't move them as it will affect current users and workflows as I am not sure if a specific run expects scripts to be in a specific location. @nuclearcat also flagged that maybe Jenkins workflow looks into `scripts/` directory for specific executables.

**Comment by yurinnick** [comment]
I am not really opinionated about this, but it can be solved with `mv` + symlink in this case.

**Comment by nuclearcat** [comment]
Small problem on staging:
```
Applying patch: patches/kernelci-core/staging.kernelci.org/0001-STAGING-add-build-configs-staging.yaml.patch
Applying patch: patches/kernelci-core/staging.kernelci.org/0002-STAGING-ensure-no-public-bisection-email-reports-are.patch
error: scripts/kci-bisect-push-results: wrong type
hint: Use 'git am --show-current-patch=diff' to see the failed patch
WARNING: Failed to apply patch
Aborting, all patches must apply.
```
adding skip label temporary to test other PR

**Comment by JenySadadia** [comment]
> Small problem on staging:
> 
> ```
> Applying patch: patches/kernelci-core/staging.kernelci.org/0001-STAGING-add-build-configs-staging.yaml.patch
> Applying patch: patches/kernelci-core/staging.kernelci.org/0002-STAGING-ensure-no-public-bisection-email-reports-are.patch
> error: scripts/kci-bisect-push-results: wrong type
> hint: Use 'git am --show-current-patch=diff' to see the failed patch
> WARNING: Failed to apply patch
> Aborting, all patches must apply.
> ```
> 
> adding skip label temporary to test other PR

Thanks for reporting. I am taking a look.

**Comment by JenySadadia** [comment]
Added `kernelci/script/update_sys_path.py` to enable imports from `kernelci` directory in `kci` scripts.

**Comment by nuclearcat** [comment]
It still fails with same error, will add staging-skip as i'm debugging also problem with my patch on kubernetes

**Comment by nuclearcat** [comment]
This is how it looks when it works:
```
Applying patch: patches/kernelci-core/staging.kernelci.org/0001-STAGING-add-build-configs-staging.yaml.patch
Applying patch: patches/kernelci-core/staging.kernelci.org/0002-STAGING-ensure-no-public-bisection-email-reports-are.patch
Applying patch: patches/kernelci-core/staging.kernelci.org/0003-STAGING-use-kernelci-staging-qemu-Docker-image.patch
Applying patch: patches/kernelci-core/staging.kernelci.org/0004-STAGING-use-staging.kernelci.org-branch-for-test-def.patch
Applying patch: patches/kernelci-core/staging.kernelci.org/0005-STAGING-use-staging-branch-for-buildroot-with-reduce.patch
Applying patch: patches/kernelci-core/staging.kernelci.org/0006-STAGING-use-staging-JWT-in-kubernetes-template.patch
```

**Comment by JenySadadia** [comment]
> It still fails with same error, will add staging-skip as i'm debugging also problem with my patch on kubernetes

Yes, I haven't fixed it yet. I'm stuck with other issues in the PR.

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+  "pydantic==1.10.5",
+  "pyelftools==0.29",
+  "pytest==7.4.0",
+  "pyyaml==6.0",
+  "requests==2.31.0",
+  "scp==0.14.5",
+  "toml==0.10.2",
+]
+authors = [
+  {name = "Guillaume Tucker" , email = "guillaume.tucker@collabora.com"},
```
Comment (gctucker) [inline pyproject.toml:35]: Please change to `gtucker@gtucker.io`, this address is not in use anymore.

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+  {name = "Matt Hart" , email = "matthew.hart@linaro.org"},
+  {name = "Jeny Sadadia" , email = "jeny.sadadia@collabora.com"},
+  {name = "Quentin Schulz" , email = "quentin.schulz@free-electrons.com"},
+  {name = "Matt Hart" , email = "github@blacklabsystems.com"},
+  {name = "Michal Galka" , email = "michal.galka@collabora.com"},
+  {name = "Corentin LABBE" , email = "clabbe@baylibre.com"},
+  {name = "Anders Roxell" , email = "anders.roxell@linaro.org"},
+  {name = "Ana Guerrero Lopez" , email = "ana.guerrero@collabora.com"},
+  {name = "Nícolas F. R. A. Prado" , email = "nfraprado@collabora.com"},
+  {name = "kernelci.org bot" , email = "bot@kernelci.org"},
```
Comment (gctucker) [inline pyproject.toml:49]: Please drop this, the bot isn't an actual author.  Maybe listing the TSC group would be useful.

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+]
+authors = [
+  {name = "Guillaume Tucker" , email = "guillaume.tucker@collabora.com"},
+  {name = "Tyler Baker" , email = "tyler.baker@linaro.org"},
+  {name = "Denys Fedoryshchenko" , email = "denys.f@collabora.com"},
+  {name = "Mark Brown" , email = "broonie@kernel.org"},
+  {name = "Kevin Hilman" , email = "khilman@baylibre.com"},
+  {name = "Matt Hart" , email = "matthew.hart@linaro.org"},
+  {name = "Jeny Sadadia" , email = "jeny.sadadia@collabora.com"},
+  {name = "Quentin Schulz" , email = "quentin.schulz@free-electrons.com"},
```
Comment (gctucker) [inline pyproject.toml:42]: This one could probably be dropped, or updated with a bootlin address.  The free-electrons.com domain doesn't exist anymore (following a trademark dispute with Free ISP iirc around 2020).

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+  {name = "Matt Hart" , email = "github@blacklabsystems.com"},
+  {name = "Michal Galka" , email = "michal.galka@collabora.com"},
+  {name = "Corentin LABBE" , email = "clabbe@baylibre.com"},
+  {name = "Anders Roxell" , email = "anders.roxell@linaro.org"},
+  {name = "Ana Guerrero Lopez" , email = "ana.guerrero@collabora.com"},
```
Comment (gctucker) [inline pyproject.toml:47]: This one can probably be dropped too, at least the email isn't valid anymore.

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+  {name = "Nícolas F. R. A. Prado" , email = "nfraprado@collabora.com"},
+  {name = "kernelci.org bot" , email = "bot@kernelci.org"},
+  {name = "Tyler Baker" , email = "forcedinductionz@gmail.com"},
+  {name = "Laura Nao" , email = "laura.nao@collabora.com"},
+  {name = "Miguel Ojeda" , email = "ojeda@kernel.org"},
+  {name = "khilman" , email = "khilman@users.noreply.github.com"},
```
Comment (gctucker) [inline pyproject.toml:53]: This looks like a dummy one that should be removed.

File: pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+  {name = "Michael Grzeschik" , email = "m.grzeschik@pengutronix.de"},
+  {name = "Alexandra Pereira" , email = "alexandra.pereira@collabora.com"},
+  {name = "Alice Ferrazzi" , email = "alice.ferrazzi@miraclelinux.com"},
+  {name = "Adrian Ratiu" , email = "adrian.ratiu@collabora.com"},
+  {name = "Chris Paterson" , email = "chris.paterson2@renesas.com"},
+  {name = "Guillaume Tucker" , email = "gtucker@gtucker.io"},
```
Comment (gctucker) [inline pyproject.toml:59]: Ah I see my current email address is here so the other @collabora.com entry can just be removed.

File: pyproject.toml [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+  {name = "Lakshmipathi" , email = "lakshmipathi.ganapathi@collabora.com"},
+  {name = "Lakshmipathi Ganapathi" , email = "lakshmipathi.ganapathi@collabora.com"},
```
Comment (gctucker) [inline pyproject.toml:61]: Ditto - email doesn't exist anymore.

File: pyproject.toml [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+  {name = "Khouloud Touil" , email = "ktouil@baylibre.com"},
+  {name = "Paweł Wieczorek" , email = "pawiecz@collabora.com"},
+  {name = "Enric Balletbo i Serra" , email = "enric.balletbo@collabora.com"},
```
Comment (gctucker) [inline pyproject.toml:64]: Ditto - email doesn't exist anymore.

File: pyproject.toml [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+  {name = "Nicolas Belin" , email = "nbelin@baylibre.com"},
+  {name = "Dan Rue" , email = "dan.rue@linaro.org"},
+  {name = "Milo Casagrande" , email = "milo.casagrande@linaro.org"},
+  {name = "Tomeu Vizoso" , email = "tomeu.vizoso@collabora.com"},
+  {name = "Aishwarya TCV" , email = "aishwarya.tcv@arm.com"},
+  {name = "Corentin Labbe" , email = "clabbe@baylibre.com"},
+  {name = "Ricardo Cañuelo" , email = "ricardo.canuelo@collabora.com"},
+  {name = "lakshmipathi.g" , email = "lakshmipathi.ganapathi@collabora.com"},
```
Comment (gctucker) [inline pyproject.toml:72]: Ditto - email doesn't exist anymore.

File: pyproject.toml [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+  {name = "Guillaume Tucker" , email = "gtucker@gtucker.io"},
+  {name = "Lakshmipathi" , email = "lakshmipathi.ganapathi@collabora.com"},
+  {name = "Lakshmipathi Ganapathi" , email = "lakshmipathi.ganapathi@collabora.com"},
+  {name = "Khouloud Touil" , email = "ktouil@baylibre.com"},
+  {name = "Paweł Wieczorek" , email = "pawiecz@collabora.com"},
+  {name = "Enric Balletbo i Serra" , email = "enric.balletbo@collabora.com"},
+  {name = "Nicolas Belin" , email = "nbelin@baylibre.com"},
+  {name = "Dan Rue" , email = "dan.rue@linaro.org"},
+  {name = "Milo Casagrande" , email = "milo.casagrande@linaro.org"},
+  {name = "Tomeu Vizoso" , email = "tomeu.vizoso@collabora.com"},
```
Comment (gctucker) [inline pyproject.toml:68]: Ditto - email doesn't exist anymore.

File: pyproject.toml [hunk 11]
```diff
@@ -0,0 +1,195 @@
...
+  {name = "Aishwarya TCV" , email = "aishwarya.tcv@arm.com"},
+  {name = "Corentin Labbe" , email = "clabbe@baylibre.com"},
+  {name = "Ricardo Cañuelo" , email = "ricardo.canuelo@collabora.com"},
+  {name = "lakshmipathi.g" , email = "lakshmipathi.ganapathi@collabora.com"},
+  {name = "Denis Efremov" , email = "efremov@linux.com"},
+  {name = "Jeny Sadadia" , email = "jeny.sadadia@gmail.com"},
+  {name = "Lakshmipathi Ganapathi" , email = "lakshmipathi.ganapathi@collabora.co.uk"},
```
Comment (gctucker) [inline pyproject.toml:75]: Ditto - email doesn't exist anymore.

File: pyproject.toml [hunk 12]
```diff
@@ -0,0 +1,195 @@
...
+  {name = "Muhammad Usama Anjum" , email = "usama.anjum@collabora.com"},
+  {name = "Lee Jones" , email = "lee.jones@linaro.org"},
+  {name = "Tzung-Bi Shih" , email = "tzungbi@kernel.org"},
+  {name = "AngeloGioacchino Del Regno" , email = "angelogioacchino.delregno@collabora.com"},
+  {name = "Florent Jacquet" , email = "florent.jacquet@free-electrons.com"},
+  {name = "Jan Luebbe" , email = "jlu@pengutronix.de"},
+  {name = "Michael Walle" , email = "michael@walle.cc"},
+  {name = "Milo Casagrande" , email = "milo@ubuntu.com"},
```
Comment (gctucker) [inline pyproject.toml:83]: Ditto - email doesn't exist anymore.

File: pyproject.toml [hunk 13]
```diff
@@ -0,0 +1,195 @@
...
+  {name = "Arisu Tachibana" , email = "arisu.tachibana@miraclelinux.com"},
+  {name = "Ed Smith" , email = "ed.smith@collabora.com"},
+  {name = "Heiko Thiery" , email = "heiko.thiery@gmail.com"},
+  {name = "Sjoerd Simons" , email = "sjoerd@collabora.com"},
+  {name = "Alex P" , email = "alexandra.pereira@collabora.com"},
```
Comment (gctucker) [inline pyproject.toml:88]: Ditto - email doesn't exist anymore.

File: pyproject.toml [hunk 14]
```diff
@@ -0,0 +1,195 @@
...
+  {name = "Detlev Casanova" , email = "detlev.casanova@collabora.com"},
+  {name = "Guillaume Tucker" , email = "guillaume@mangoz.org"},
```
Comment (gctucker) [inline pyproject.toml:90]: Ha, that's a weird one.  I don't think I've used this address in about 10 years :p  So please drop it as well.

File: pyproject.toml [hunk 15]
```diff
@@ -0,0 +1,195 @@
...
+  {name = "Jerome Brunet" , email = "jbrunet@baylibre.com"},
+  {name = "Nick Desaulniers" , email = "ndesaulniers@google.com"},
+  {name = "Ricardo Ribalda" , email = "ribalda@chromium.org"},
+  {name = "Chase Qi" , email = "chase.qi@linaro.org"},
+  {name = "Florian Bezdeka" , email = "florian.bezdeka@siemens.com"},
+  {name = "Kevin Hilman" , email = "khilman@kernel.org"},
+  {name = "Kevin Hilman" , email = "khilman@linaro.org"},
```
Comment (gctucker) [inline pyproject.toml:97]: Probably each author should only appear once in this list.  It's not a list of email addresses but a list of people currently contributing to the code.

File: pyproject.toml [hunk 16]
```diff
@@ -0,0 +1,195 @@
...
+  {name = "Lakshmipathi.G" , email = "lakshmipathi.ganapathi@collabora.com"},
```
Comment (gctucker) [inline pyproject.toml:98]: Ditto - email doesn't exist anymore.

File: pyproject.toml [hunk 17]
```diff
@@ -0,0 +1,195 @@
...
+  {name = "Chris Paterson" , email = "chris.paterson2@renesas.com"},
+  {name = "Guillaume Tucker" , email = "gtucker@gtucker.io"},
+  {name = "Lakshmipathi" , email = "lakshmipathi.ganapathi@collabora.com"},
+  {name = "Lakshmipathi Ganapathi" , email = "lakshmipathi.ganapathi@collabora.com"},
+  {name = "Khouloud Touil" , email = "ktouil@baylibre.com"},
+  {name = "Paweł Wieczorek" , email = "pawiecz@collabora.com"},
+  {name = "Enric Balletbo i Serra" , email = "enric.balletbo@collabora.com"},
+  {name = "Nicolas Belin" , email = "nbelin@baylibre.com"},
+  {name = "Dan Rue" , email = "dan.rue@linaro.org"},
+  {name = "Milo Casagrande" , email = "milo.casagrande@linaro.org"},
```
Comment (gctucker) [inline pyproject.toml:67]: Ditto - email doesn't exist anymore (foundries.io is his current address)

File: pyproject.toml [hunk 18]
```diff
@@ -0,0 +1,195 @@
...
+  {name = "Mark Brown" , email = "broonie@sirena.org.uk"},
+  {name = "Matt Hart" , email = "matt@blacklabsystems.com"},
+  {name = "Matt Hart" , email = "matt@mattface.org"},
+  {name = "Mattijs Korpershoek" , email = "mkorpershoek@baylibre.com"},
+  {name = "Milo Casagrande" , email = "milo@opensourcefoundries.com"},
+  {name = "Suram Chandra Sekhar" , email = "suram@nxp.com"},
+  {name = "Suram Chandra Sekhar" , email = "suram@oss.nxp.com"},
+  {name = "Tyler Baker" , email = "tyler@opensourcefoundries.com"},
+  {name = "William K Lin" , email = "lin.wen.kuang@gmail.com"},
+  {name = "alan bennett" , email = "akbennett@hasdell"},
```
Comment (gctucker) [inline pyproject.toml:109]: Ditto - email doesn't exist anymore.

File: pyproject.toml [hunk 19]
```diff
@@ -0,0 +1,195 @@
...
+  {name = "isaiahduncan" , email = "32970507+isaiahduncan@users.noreply.github.com"},
```
Comment (gctucker) [inline pyproject.toml:110]: Dummy entry.

File: pyproject.toml [hunk 20]
```diff
@@ -0,0 +1,195 @@
...
+  {name = "lollivier" , email = "lollivier@baylibre.com"},
+  {name = "Alexandra" , email = "alexandra@arch.localhost"},
```
Comment (gctucker) [inline pyproject.toml:112]: Dummy entry.

File: pyproject.toml [hunk 21]
```diff
@@ -0,0 +1,195 @@
...
+  {name = "Alexandre Belloni" , email = "alexandre.belloni@free-electrons.com"},
+  {name = "Alice Ferrazzi" , email = "alice.ferrazzi@cybertrust.co.jp"},
```
Comment (gctucker) [inline pyproject.toml:114]: Replaced with Arisu.

File: pyproject.toml [hunk 22]
```diff
@@ -0,0 +1,195 @@
...
+  {name = "Bastian Krause" , email = "bst@pengutronix.de"},
+  {name = "Bjorn Andersson" , email = "bjorn.andersson@linaro.org"},
+  {name = "Denys Fedoryshchenko" , email = "nuclearcat@nuclearcat.com"},
+  {name = "Fathi Boudra" , email = "fathi.boudra@linaro.org"},
+  {name = "Gustavo Padovan" , email = "gustavo.padovan@collabora.com"},
+  {name = "Heidi Fahim" , email = "heidifahim@google.com"},
+  {name = "Jakob Unterwurzacher" , email = "jakob.unterwurzacher@theobroma-systems.com"},
+  {name = "Jan-Simon Möller" , email = "dl9pf@gmx.de"},
+  {name = "Kees Cook" , email = "keescook@chromium.org"},
+  {name = "Lakshmipathi.G" , email = "Lakshmipathi.Ganapathi@Collabora.com"},
```
Comment (gctucker) [inline pyproject.toml:127]: Ditto - email doesn't exist anymore.

It's impressive how many email addresses Laks has used :)

File: pyproject.toml [hunk 23]
```diff
@@ -0,0 +1,195 @@
...
+  {name = "Matt Hart" , email = "matthew.hart@arm.com"},
+  {name = "Matthias Maennich" , email = "maennich@google.com"},
+  {name = "Maxime Ripard" , email = "maxime.ripard@free-electrons.com"},
```
Comment (gctucker) [inline pyproject.toml:130]: Ditto - email doesn't exist anymore.

File: pyproject.toml [hunk 24]
```diff
@@ -0,0 +1,195 @@
...
+  {name = "Mylène Josserand" , email = "mylene.josserand@collabora.com"},
```
Comment (gctucker) [inline pyproject.toml:131]: Ditto - email doesn't exist anymore.

File: pyproject.toml [hunk 25]
```diff
@@ -0,0 +1,195 @@
...
+  {name = "Neil Williams" , email = "codehelp@debian.org"},
+  {name = "Nick Desaulniers" , email = "nickdesaulniers@users.noreply.github.com"},
```
Comment (gctucker) [inline pyproject.toml:133]: Dummy entry.

File: pyproject.toml [hunk 26]
```diff
@@ -0,0 +1,195 @@
...
+  {name = "Nikolai Kondrashov" , email = "Nikolai.Kondrashov@redhat.com"},
+  {name = "Nikolay Yurin" , email = "yurinnick@meta.com"},
+  {name = "Quentin Schulz" , email = "q.schulz@gmail.com"},
+  {name = "Ricardo Ribalda" , email = "ricardo@ribalda.com"},
+  {name = "Riku Voipio" , email = "riku.voipio@linaro.org"},
+  {name = "Robin Lu" , email = "robin.lubin@shopee.com"},
+  {name = "Sasha Levin" , email = "sashal@kernel.org"},
+  {name = "Shreeya Patel" , email = "shreeya.patel@collabora.com"},
+  {name = "Sjoerd Simons" , email = "sjoerd.simons@collabora.co.uk"},
+  {name = "Thomas Petazzoni" , email = "thomas.petazzoni@free-electrons.com"},
```
Comment (gctucker) [inline pyproject.toml:143]: Ditto - email doesn't exist anymore.

File: pyproject.toml [hunk 27]
```diff
@@ -0,0 +1,195 @@
...
+  {name = "Vince Hillier" , email = "vince@revenni.com"},
+  {name = "Wang Mingyu" , email = "wangmy@cn.fujitsu.com"},
+  {name = "nkrishna" , email = "nkrishna@otc-prt.org"},
+  {name = "roxell" , email = "anders.roxell@gmail.com"},
+  {name = "sjoerd-ccu" , email = "sjoerd.simons@collabora.co.uk"}
```
Comment (gctucker) [inline pyproject.toml:148]: There's already a Sjoerd Simmons entry so this one should be dropped.

File: pyproject.toml [hunk 28]
```diff
@@ -0,0 +1,195 @@
...
+]
+maintainers = [
+  {name = "Denys Fedoryshchenko", email = "denys.f@collabora.com"},
+  {name = "Jeny Sadadia", email = "jeny.sadadia@collabora.com"}
+]
```
Comment (gctucker) [inline pyproject.toml:153]: Arisu is listed as co-maintainer too: https://kernelci.org/docs/org/maintainers/#core-tools

**Review by gctucker — CHANGES_REQUESTED** [review]
A number of entries in the `authors` list aren't valid.  I think this is merely for users who install the package to know who is currently contributing to the code rather than a legally-binding list of contributors, which is covered by the headers in each file in the source code.  So probably it would be better to look at the top contributors or people who have contributed in the past year to manually curate this list into a more representative one.

**Comment by JenySadadia** [comment]
> A number of entries in the `authors` list aren't valid. I think this is merely for users who install the package to know who is currently contributing to the code rather than a legally-binding list of contributors, which is covered by the headers in each file in the source code. So probably it would be better to look at the top contributors or people who have contributed in the past year to manually curate this list into a more representative one.

I have used `git shortlog --summary --email` to get all the developers and added them to this list.
Checking every file manually for the commit header would take a lot of time.

**Comment by JenySadadia** [comment]
> > > > A number of entries in the `authors` list aren't valid. I think this is merely for users who install the package to know who is currently contributing to the code rather than a legally-binding list of contributors, which is covered by the headers in each file in the source code. So probably it would be better to look at the top contributors or people who have contributed in the past year to manually curate this list into a more representative one.
> > > 
> > > 
> > > I have used `git shortlog --summary --email` to get all the developers and added them to this list. Checking every file manually for the commit header would take a lot of time.
> > 
> > 
> > I don't think that's a valid approach, having this inaccurate list of contributors isn't helpful. I don't think it should take more than 15min to prepare by hand, in fact if you apply the changes from the PR review that would already be mostly done. I would suggest to not have a list of contributors at all if it's just a dump of the git history which isn't representative and contains lots of stale entries.
> 
> After taking a look at various popular Python projects, I've noticed that most of them don't specify a list of authors in `pyproject.toml`. Some have an `AUTHORS` file but that's really not required. So I think having a generic entry with the mailing list would be useful, and ensuring the list of maintainers and co-maintainers is accurate (so please add @aliceinwire to the list there as per my review comment).
> 
> It's also useful to take a look at how the legacy PyPI entry looks since this is what users will be seeing first with the v2.0 upload: https://pypi.org/project/kernelci/

Thanks for the reviews.
I have updated the list of maintainers and added Arisu to it.
About the authors, do you mean I should drop all the authors and just have `kernelci@groups.io` in the list?

**Comment by gctucker** [comment]
Yes I think the list of authors should be dropped, and the new mailing list `kernelci@lists.linux.dev` should be mentioned somewhere in the meta-data but not necessarily as "authors".  One option is to just have this I guess:
```toml
authors = [
  {name = "KernelCI", email = "kernelci@lists.linux.dev"},
]
```
as I don't see any dedicated fields for a mailing list.  Django has something similar, except their have their own dedicated foundation:
https://pypi.org/project/Django/

**Comment by JenySadadia** [comment]
> Yes I think the list of authors should be dropped, and the new mailing list `kernelci@lists.linux.dev` should be mentioned somewhere in the meta-data but not necessarily as "authors". One option is to just have this I guess:
> 
> ```toml
> authors = [
>   {name = "KernelCI", email = "kernelci@lists.linux.dev"},
> ]
> ```
> 
> as I don't see any dedicated fields for a mailing list. Django has something similar, except their have their own dedicated foundation: https://pypi.org/project/Django/

Okay. Thanks for the comment. Updated the authors' list.

File: pyproject.toml [hunk 29]
```diff
@@ -0,0 +1,83 @@
...
+
+[project.scripts]
+kci = "kernelci.scripts.kci:main"
+kci_bisect = "kernelci.scripts.kci_bisect:main"
+kci_build = "kernelci.scripts.kci_build:main"
+kci_data = "kernelci.scripts.kci_data:main"
+kci_rootfs = "kernelci.scripts.kci_rootfs:main"
+kci_test = "kernelci.scripts.kci_test:main"
+kci_bisect_lava_v2_callback = "kernelci.scripts.kci_bisect_lava_v2_callback:main"
+kci_bisect_push_results = "kernelci.scripts.kci_bisect_push_results:main"
```
[the remark this answers is not in the document]
Comment (JenySadadia) [reply pyproject.toml:83]: That's exactly why we have symlinks in the current location of the scripts e.g. `scripts/kci-bisect-push-results` 
It wasn't possible to use script names with `-` in the `project.scripts` sections. That's why I needed to rename the scripts.

[the remark this answers is not in the document]
Comment (gctucker) [reply pyproject.toml:83]: Ah I see, well I didn't notice the symlinks.  Still I wonder if these scripts should actually be made part of the package, but I guess there isn't a clear deprecation timeline yet.

Alright so as long as Jenkins jobs aren't broken then it should be fine.  Let's just hope that moving to symlinks doesn't have unwanted side effects.

[the remark this answers is not in the document]
Comment (JenySadadia) [reply pyproject.toml:83]: Tried to run Jenkins job https://bot.staging.kernelci.org/job/test-bisect/3/.
It wasn't able to kind `kci-bisect-lava-v2-callback` script. 
```
+ PYTHON_EGG_CACHE=/data/workspace/bot.staging.kernelci.org/kernel-builder/workspace/test-bisect/python-egg-cache kci-bisect-lava-v2-callback --token=**** --test-case=baseline.dmesg.emerg callback.json
/data/workspace/bot.staging.kernelci.org/kernel-builder/workspace/test-bisect__tmp/durable-171833ff/script.sh: 2: kci-bisect-lava-v2-callback: not found
```
Updated the PR to fix the issue.

[the remark this answers is not in the document]
Comment (JenySadadia) [reply pyproject.toml]: Tested OK: https://bot.staging.kernelci.org/job/test-bisect/6/

**Comment by nuclearcat** [comment]
Thanks Jeny, tested on staging, and thanks for @gctucker for hint to possible point of failure and review.

**Pull request merged**